cfme/networks/topology.py

```diff
@@ -0,0 +1,222 @@
+
+    def set_active(self, active=True):
+        if active != self.is_active:
+            self.element.click()
+
+
+@attr.s
+class TopologyLegendCollection(BaseCollection):
+    """Collection object for legends in topology"""
+    ENTITY = TopologyLegend
+
+    def all(self, topology):
+        final_legends = []
+        legends = topology.browser.elements(topology.view.LEGENDS)
+        for element in legends:
+            legend_name = topology.browser.text(element.find_element_by_tag_name('label'))
+            final_legends.append(self.instantiate(element, legend_name))
+        return final_legends
+
+
+@attr.s
+class TopologyDisplayNames(BaseEntity):
+    element = attr.ib()
+
+    @property
+    def is_enabled(self):
+        return self.element.is_selected()
+
+    def enable(self, enable=True):
+        if self.is_enabled != enable:
+            self.element.click()
+
+    def disable(self):
+        self.enable(enable=False)
+
+
+@attr.s
+class TopologyElement(BaseEntity):
+    element = attr.ib()
+    obj = attr.ib()
+
+    def refresh(self):
+        if self.element is None:
+            raise ValueError('Element should not be None')
+        element_data = re.search('Name: (.*) Type: (.*) Status: (.*)', self.element.text)
+        if len(element_data.groups()) != 3:
+            raise RuntimeError('Topology element does not contain name, type or status')
+        self.name = element_data.group(1)
+        self.type = element_data.group(2)
+        self.status = element_data.group(3)
+        self.x = round(float(self.element.get_attribute('cx')), 1)
+        self.y = round(float(self.element.get_attribute('cy')), 1)
+
+    def __repr__(self):
+        return "<TopologyElement name:{}, type:{}, status:{}, x:{}, y:{}, is_hidden:{}>".format(
+            self.name, self.type, self.status, self.x, self.y, self.is_hidden)
+
+    @property
+    def is_hidden(self):
+        return 'opacity: 0.2' in self.element.get_attribute('style')
+
+    @property
+    def parents(self):
+        elements_obj = []
+        for lines_obj in [lines_obj for lines_obj in self.object.lines_obj()
+                          if lines_obj.x2 == self.x and lines_obj.y2 == self.y]:
+                for element in self.object.elements_obj():
+                    if element.x == lines_obj.x1 and element.y == lines_obj.y1:
+                        elements_obj.append(element)
+        return elements_obj
+
+    @property
+    def children(self):
+        elements_obj = []
+        for lines_obj in [lines_obj for lines_obj in self.object.lines_obj()
+                          if lines_obj.x1 == self.x and lines_obj.y1 == self.y]:
+            for element in self.object.elements_obj():
+                if element.x == lines_obj.x2 and element.y == lines_obj.y2:
+                    elements_obj.append(element)
+        return elements_obj
+
+    def double_click(self):
+        self.obj.browser.double_click(self.element)
+
+    def is_displayed(self):
```

Comment:
**REQUIRED**
Why is it here and not in the view? Also `is_displayed` is supposed to be a property.

Reply:
@PatrikH0lop here is still not fixed